packages/renderer/src/lib/image/PushManifestModal.spec.ts

```diff
@@ -39,23 +39,14 @@ vi.mock('@xterm/xterm', () => {
   };
 });
 
-const getConfigurationValueMock = vi.fn();
-const pushManifestMock = vi.fn();
-
 beforeAll(() => {
   (window.events as unknown) = {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     receive: (_channel: string, func: any) => {
       func();
     },
   };
-  (window as any).ResizeObserver = vi.fn().mockReturnValue({ observe: vi.fn(), unobserve: vi.fn() });
-  (window as any).getImageInspect = vi.fn().mockImplementation(() => Promise.resolve({}));
-  (window as any).logsContainer = vi.fn().mockResolvedValue(undefined);
-  (window as any).refreshTerminal = vi.fn();
-  (window as any).getConfigurationValue = getConfigurationValueMock;
-  (window as any).showMessageBox = vi.fn();
-  (window as any).pushManifest = pushManifestMock;
+  (window.dispatchEvent as unknown) = vi.fn();
```

Comment:
is it required ? all window methods should be created automatically

wondering if you just need a spy

const dispatchEventSpy = spy.....(window, 'dispatchEvent')

also (window as unknown) is deprecated over Object.defineProperty

Reply:
If I remove this line it says `TypeError: [Function bound dispatchEvent] is not a spy or a call to a spy!`; I will be replacing with `Object.defineProperty`